pkg/stream/xprotocol/connpool_binding.go

```diff
@@ -81,11 +135,16 @@ func (p *poolBinding) NewStream(ctx context.Context, receiver types.StreamReceiv
 
 	downstreamConn := getDownstreamConn(ctx)
 	c.downstreamConn = downstreamConn
-	downstreamConn.AddConnectionEventListener(downstreamCloseListener{upstreamClient: c})
+	p.listenerOnce.Do(func() {
```

Comment:
`poolBinding` has many connections，and  `listenerOnce` should work on `activeClientBinding`.